src/ralph/backends/data/mongo.py

```diff
@@ -0,0 +1,447 @@
+    COLLECTION: str = None
+    CLIENT_OPTIONS: MongoClientOptions = MongoClientOptions()
+    DEFAULT_QUERY_STRING: str = "*"
+    DEFAULT_CHUNK_SIZE: int = 500
+    LOCALE_ENCODING: str = "utf8"
+
+
+class MongoQuery(BaseQuery):
+    """Mongo query model."""
+
+    filter: Optional[dict]
+    projection: Optional[dict]
+
+
+class MongoDataBackend(BaseDataBackend):
+    """Mongo database backend."""
+
+    name = "mongo"
+    query_model = MongoQuery
+    default_operation_type = BaseOperationType.CREATE
+    settings_class = MongoDataBackendSettings
+
+    def __init__(self, settings: settings_class = None):
+        """Instantiates the Mongo client.
+
+        Args:
+            connection_uri (str): MongoDB connection URI.
+            database (str): MongoDB database to connect to.
+            collection (str): MongoDB database collection to get objects from.
+            client_options (MongoClientOptions): A dictionary of valid options
+            for the MongoClient class initialization.
+        """
+        self.client = MongoClient(
+            settings.CONNECTION_URI, **settings.CLIENT_OPTIONS.dict()
+        )
+        self.database = getattr(self.client, settings.DATABASE)
+        self.collection = getattr(self.database, settings.COLLECTION)
+        self.default_chunk_size = settings.DEFAULT_CHUNK_SIZE
+        self.locale_encoding = settings.LOCALE_ENCODING
+
+    def status(self) -> DataBackendStatus:
+        """Checks MongoDB cluster connection status."""
+        # Check Mongo cluster connection
+        try:
+            self.client.admin.command("ping")
+        except ConnectionFailure:
+            return DataBackendStatus.AWAY
+
+        # Check cluster status
+        if self.client.admin.command("serverStatus").get("ok", 0.0) < 1.0:
+            return DataBackendStatus.ERROR
+
+        return DataBackendStatus.OK
+
+    def list(
+        self, target: str = None, details: bool = False, new: bool = False
+    ) -> Iterator[Union[str, dict]]:
+        """Lists documents for a given collection.
+
+        Args:
+            details (bool): Get detailed archive information instead of just ids.
+            new (bool): Given the history, list only not already fetched archives.
+        """
+        collection = self.collection
+        if target:
+            collection = getattr(self.database, target)
+        for document in collection.find():
+            if details:
+                yield document
+            else:
+                yield str(document.get("_id"))
+
+    @enforce_query_checks
+    def read(
+        self,
+        *,
+        query: Union[str, MongoQuery] = None,
+        target: str = None,
+        chunk_size: Union[None, int] = None,
+        raw_output: bool = False,
+        ignore_errors: bool = False,
+    ) -> Iterator[Union[bytes, dict]]:
+        """Gets collection documents and yields them.
+
+        The `query` dictionary should only contain kwargs compatible with the
```

Comment:
It would be nice to use the same template for this docstring as the others above/the ones from base.py